### DISCUSSION TITLE
Interface API filter for connected status

### DISCUSSION THREAD

**Discussion by ghost** [netbox-community/netbox#9782] [Q&A]
Hi Everybody!

I've been working with the API trying to filter out any free interfaces on a given device, I've looked into the cabled and connected filters and it seems that cabled is behaving as expected (filters interfaces with an actual cable connected).  
My issue is with the connected filter, it only accounts for interfaces with an active path but ignores any interface with the marked_connected value set (no actual cable object but we know it's connected to something).

Is that the expected behavior?  
As I see it connected either matches against the _occupied value as well or we should have another field for the occupied value?

API call example:
`/api/dcim/interfaces/?kind=physical&type=1000base-t&mgmt_only=false&brief=true&connected=false`

Response (truncated and redacted)
```json
{
    "count": 12801,
    "next": "****/api/dcim/interfaces/?brief=true&connected=false&kind=physical&limit=50&mgmt_only=false&offset=50&type=1000base-t",
    "previous": null,
    "results": [
        {
            "id": 49017,
            "url": "****/api/dcim/interfaces/49017/",
            "display": "ge-0/0/0",
            "device": {
                "id": 945,
                "url": "****/api/dcim/devices/945/",
                "display": "****tor-01",
                "name": "****tor-01"
            },
            "name": "ge-0/0/0",
            "cable": null,
            "_occupied": true
        },
...
```

**Comment by jeremystretch** [comment]
@omri-ice seems you've run into the limitation that FR #8580 hopes to solve. Please let us know if you have any input on that issue!
